### PR TITLE
Refactor add to trace

### PR DESCRIPTION
**Pull request by MartinRavn**
This is a continuation of #731 which was accidentally merged. I have fixed a few bugs and updated the documentation.

### CODE REVIEW

File: NuRadioReco/framework/base_trace.py [hunk 1]
```diff
@@ -248,35 +248,36 @@ def deserialize(self, data_pkl):
...
-        t1_channel = t0_channel + channel.get_number_of_samples() / channel.get_sampling_rate()
+        t1_channel = tt_channel[-1]
         sampling_rate_channel = channel.get_sampling_rate()
         n_samples_channel = channel.get_number_of_samples()
 
         # We handle 1+2x2 cases:
-        # 1. Channel is completely outside readout window:
-        if t1_channel < t0_readout or t1_readout < t0_channel:
+        # 1. Channel is completely outside readout window (less than one full bin overlap):
+        if t1_channel < tt_readout[1] or t1_readout < t0_channel:
```
Comment (fschlueter) [inline NuRadioReco/framework/base_trace.py:280]: why was to old condition not okay?

File: NuRadioReco/framework/base_trace.py [hunk 2]
```diff
@@ -292,15 +293,15 @@ def add_to_trace(self, channel):
             t_start_channel = t0_channel
         # 4. Channel ends after readout window:
         if t1_channel >= t1_readout:
-            i_end_readout = n_samples_readout
+            i_end_readout = n_samples_readout-1
             t_end_readout = t1_readout
             i_end_channel = int( (t1_readout - t0_channel) * sampling_rate_channel ) + 1 # The bin of channel right after readout ends
```
Comment (fschlueter) [inline NuRadioReco/framework/base_trace.py:298]: would you mind removing the whitespaces: `int(bla bla)`

Comment (fschlueter) [reply NuRadioReco/framework/base_trace.py:298]: everywhere

File: NuRadioReco/framework/base_trace.py [hunk 3]
```diff
@@ -292,15 +293,15 @@ def add_to_trace(self, channel):
             t_start_channel = t0_channel
         # 4. Channel ends after readout window:
         if t1_channel >= t1_readout:
-            i_end_readout = n_samples_readout
+            i_end_readout = n_samples_readout-1
```
Comment (fschlueter) [inline NuRadioReco/framework/base_trace.py:296]: you can add whitespaces here: `n - 1`

**Comment by fschlueter** [comment]
Looks good to me. Can you fix the coding style and we merge it?

**Review by fschlueter — APPROVED** [review]

File: NuRadioReco/framework/base_trace.py (hunk 1 above)
Comment (MartinRavn) [reply NuRadioReco/framework/base_trace.py:280]: I actually don't think I need this change anymore after fixing the other bug. I'll run a test to see of the old condition causes problems.

Comment (fschlueter) [reply NuRadioReco/framework/base_trace.py:280]: What was the bug, I do not see related changes.

Comment (MartinRavn) [reply NuRadioReco/framework/base_trace.py]: There were no problems in my test so I reverted this change now.

**Comment by fschlueter** [comment]
Cool. Go ahead and merge this if you want.

**Pull request merged**